### PR TITLE
Keycloak documentation : fix variables, hostnames, java flavor

### PR DESCRIPTION
**Pull request by please-openit**
## Describe your PR

Keycloak documentation up to date with all supported variables and features

### CODE REVIEW

File: content/doc/addons/keycloak.md [hunk 1]
```diff
@@ -60,7 +60,7 @@ An add-on update might require a restart.
 
 By default, Keycloak on Clever Cloud uses small-size resources, i.e:
 
-- XS Java
+- S Java
```
Comment (allemas) [inline content/doc/addons/keycloak.md:63]: We are still in XS

**Comment by allemas** [comment]
LGTM for me except for the java size runtime.

File: content/doc/addons/keycloak.md [hunk 2]
```diff
@@ -130,23 +130,57 @@ There are two ways to export Keycloak data:
 
 2. A full export using the environment variables of the Keycloak Java application from the Clever Cloud console.
     - Go to the Keycloak Java application in the Clever Cloud console
-    - In the Java application, you can explicitly Realm you want to export as arguments to  the `CC_EXPORT_REALM` environment variable;
+    - In the Java application, you can explicitly Realms (separated each of them with `,`) you want to export as arguments to the `CC_KEYCLOAK_EXPORT_REALMS` environment variable;
```
Comment (davlgd) [inline content/doc/addons/keycloak.md:133]:
```suggestion
    - In the Java application, you can explicitly declare Realms (separated each of them with `,`) you want to export as arguments to the `CC_KEYCLOAK_EXPORT_REALMS` environment variable;
```

File: content/doc/addons/keycloak.md [hunk 3]
```diff
@@ -118,7 +118,7 @@ There are two ways to manage Realms:
 
 Keycloak uses an [FSBucket](https://developers.clever-cloud.com/doc/addons/fs-bucket/) to install themes and plugins. To deploy a custom theme or custom plugin, simply download them into the respective `themes` or `providers` folder in your FSBucket.
 
-## Exporting data
+## Exporting realms data
```
Comment (davlgd) [inline content/doc/addons/keycloak.md:121]:
```suggestion
### Exporting realms data
```

File: content/doc/addons/keycloak.md [hunk 4]
```diff
@@ -130,23 +130,57 @@ There are two ways to export Keycloak data:
...
-Uploading previously exported data in the `Import` folder in the associated FSBucket enables importing realms data. The import process starts after restarting the Java application.
+## Importing realms data
 
-### Known issues
-#### Java application deployment may fail
+Uploading previously exported data in `realms/import` folder in the associated FSBucket enables importing realms data. The import process starts after restarting the Java application.
+
+## Add IP filtering in Keycloak for admin console
+
+2 specific authentication flows with a filter based on IP Addresses we made especially for you has been created and affected as default to clients "security-admin-console" and "admin-cli".
```
Comment (davlgd) [inline content/doc/addons/keycloak.md:152]:
```suggestion
Two specific authentication flows with an IP addresses based filter are especially created and affected as default to clients "security-admin-console" and "admin-cli". To use them: 
```

File: content/doc/addons/keycloak.md [hunk 5]
```diff
@@ -130,23 +130,57 @@ There are two ways to export Keycloak data:
...
 
     > Don't forget to apply change at the bottom of the environment variable page if you use the console.
 
 > This is a total export, including the client's secrets and hashed password
 
-### Importing realms data
+Your exported realm is available in `realms/export` folder in the associated FSBucket.
 
-Uploading previously exported data in the `Import` folder in the associated FSBucket enables importing realms data. The import process starts after restarting the Java application.
+## Importing realms data
```
Comment (davlgd) [inline content/doc/addons/keycloak.md:146]: I think we should place this in the `Realms managment` section and move plugins/themes stuff outside of it. 

```suggestion
### Importing realms data
```

File: content/doc/addons/keycloak.md [hunk 6]
```diff
@@ -130,23 +130,57 @@ There are two ways to export Keycloak data:
...
+## Add IP filtering in Keycloak for admin console
+
+2 specific authentication flows with a filter based on IP Addresses we made especially for you has been created and affected as default to clients "security-admin-console" and "admin-cli".
+
+First, enable "PLEASE-OPEN.IT Authenticator IP Range" to "Required".
+
+Then, click on the crank to access to parameters and put authorized IPs to access to resources.
+
+**Do not forget to make the same on each realm you want to protect.**
+
```
Comment (davlgd) [inline content/doc/addons/keycloak.md:159]:
```suggestion
- Enable "PLEASE-OPEN.IT Authenticator IP Range" to "Required".
- Click on the crank to access parameters
- Set IPs with authorized access

**Do not forget to make the same on each realm you want to protect.**

```

File: content/doc/addons/keycloak.md [hunk 7]
```diff
@@ -130,23 +130,57 @@ There are two ways to export Keycloak data:
...
+Those flows could be affected to your own clients if you need.
+
+## Hostnames
+
+By default, your Keycloak instance is expose through a Clever-Cloud addon domain, avaible in the addon dashboard.
```
Comment (davlgd) [inline content/doc/addons/keycloak.md:164]:
```suggestion
By default, your Keycloak instance is exposed through a Clever Cloud service domain, as mentioned in the `Service dependencies` tab of your add-on.
```

File: content/doc/addons/keycloak.md [hunk 8]
```diff
@@ -130,23 +130,57 @@ There are two ways to export Keycloak data:
...
+
+### Custom hostname
+
+In your Java runtime, add your own domains to the "domains" section ([https://developers.clever-cloud.com/doc/administrate/domain-names/](https://developers.clever-cloud.com/doc/administrate/domain-names/)).
+
+Change the variable `CC_KEYCLOAK_HOSTNAME`, then restart your Java runtime.
```
Comment (davlgd) [inline content/doc/addons/keycloak.md:170]:
```suggestion
You can use your own domain. [Just set it](/doc/administrate/domain-names/) in the `Domains` section of the Java application of the Keycloak add-on. Then, edit the `CC_KEYCLOAK_HOSTNAME` environment variable, apply this change and restart the application.
```

File: content/doc/addons/keycloak.md [hunk 9]
```diff
@@ -130,23 +130,57 @@ There are two ways to export Keycloak data:
...
+
+> Don’t forget to apply change at the bottom of the environment variable page if you use the console.
+
+### Admin hostname
+
+Keycloak can expose administration console on a different hostname. By default, we use the same hostname, define `CC_KEYCLOAK_HOSTNAME_ADMIN` variable then restart your Java runtime to get a specific hostname admin.
```
Comment (davlgd) [inline content/doc/addons/keycloak.md:176]:
```suggestion
By default, Keycloak expose administration console on the same hostname but [you can use another one](https://www.keycloak.org/server/hostname#_exposing_the_administration_console_on_a_separate_hostname). Add its domain to the application and set it through the `CC_KEYCLOAK_HOSTNAME_ADMIN` environment variable. Then apply this change and restart the application.
```

File: content/doc/addons/keycloak.md [hunk 10]
```diff
@@ -130,23 +130,57 @@ There are two ways to export Keycloak data:
...
+
+By default, your Keycloak instance is expose through a Clever-Cloud addon domain, avaible in the addon dashboard.
+
+### Custom hostname
+
+In your Java runtime, add your own domains to the "domains" section ([https://developers.clever-cloud.com/doc/administrate/domain-names/](https://developers.clever-cloud.com/doc/administrate/domain-names/)).
+
+Change the variable `CC_KEYCLOAK_HOSTNAME`, then restart your Java runtime.
+
+> Don’t forget to apply change at the bottom of the environment variable page if you use the console.
```
Comment (davlgd) [inline content/doc/addons/keycloak.md:172]: This is more a Clever Cloud Console rule, not specific to Keycloak add-on

```suggestion
```

File: content/doc/addons/keycloak.md [hunk 11]
```diff
@@ -130,23 +130,57 @@ There are two ways to export Keycloak data:
...
+
+### Admin hostname
+
+Keycloak can expose administration console on a different hostname. By default, we use the same hostname, define `CC_KEYCLOAK_HOSTNAME_ADMIN` variable then restart your Java runtime to get a specific hostname admin.
+
+> Don’t forget to apply change at the bottom of the environment variable page if you use the console.
```
Comment (davlgd) [inline content/doc/addons/keycloak.md:178]:
```suggestion
```

File: content/doc/addons/keycloak.md [hunk 12]
```diff
@@ -130,23 +130,57 @@ There are two ways to export Keycloak data:
...
+
+[https://www.keycloak.org/server/hostname#_exposing_the_administration_console_on_a_separate_hostname](https://www.keycloak.org/server/hostname#_exposing_the_administration_console_on_a_separate_hostname)
```
Comment (davlgd) [inline content/doc/addons/keycloak.md:180]:
```suggestion
```

File: content/doc/addons/keycloak.md [hunk 13]
```diff
@@ -91,7 +91,7 @@ Refer to the [Clever Tools documentation](https://github.com/CleverCloud/clever-
 
 Once you have created your add-on, go to the Console Keycloak Java runtime page and click "Open the application". You also can get the Keycloak add-on URL from the Clever Cloud console, in the information panel of your Keycloak add-on.
 
-The admin interface is accessible on the `/admin` endpoint. The default user and password are provided in the Java application by the environment variables `CC_KEYCLOAK_ADMIN` and `CC_KEYCLOAK_ADMIN_PASSWORD`. A password change is requested at the first connection.
+The admin interface is accessible on the `/admin` endpoint. The default user and password are provided in the Java application by the environment variables `CC_KEYCLOAK_ADMIN` and `CC_KEYCLOAK_ADMIN_DEFAULT_PASSWORD`. A password change is requested at the first connection.
```
Comment (davlgd) [inline content/doc/addons/keycloak.md:94]:
```suggestion
The admin interface is accessible on the `/admin` endpoint. The default user and password are provided in the Java application by the environment variables `CC_KEYCLOAK_ADMIN` and `CC_KEYCLOAK_ADMIN_DEFAULT_PASSWORD`. 

A password change is requested at the first connection.
```

File: content/doc/addons/keycloak.md [hunk 14]
```diff
@@ -130,23 +130,57 @@ There are two ways to export Keycloak data:
...
+        `CC_KEYCLOAK_EXPORT_REALMS=my_realm`
 
     - restart the Java application from the Clever Cloud console.
 
     > Don't forget to apply change at the bottom of the environment variable page if you use the console.
 
 > This is a total export, including the client's secrets and hashed password
 
-### Importing realms data
+Your exported realm is available in `realms/export` folder in the associated FSBucket.
```
Comment (davlgd) [inline content/doc/addons/keycloak.md:144]:
```suggestion
Your exported realm is available in `realms/export` folder in the associated FS Bucket.
```

**Review by davlgd — CHANGES_REQUESTED** [review]
Some edits, if you're ok with this, I'll check the final layout and merge

**Pull request merged**